scipy/_lib/_array_api.py

```diff
@@ -326,6 +358,8 @@ def xp_assert_close(actual, desired, *, rtol=None, atol=0, check_namespace=True,
     __tracebackhide__ = True  # Hide traceback for py.test
     if xp is None:
         xp = array_namespace(actual)
+    if isinstance(desired, list):
+        desired = xp.asarray(desired, dtype=actual.dtype)
```

Comment:
Hmm, I needed a patch like below to "fix" some failing cases locally for array API suite, which seem to show up outside of the array API suite in the CI as you say.

I'm a bit nervous about this additional patch tough--are there really cases where both `actual` and `desired` should be array-likes/lists in array API context? Or is the real problem a need for some additional skips/filters, etc.

```diff
--- a/scipy/_lib/_array_api.py
+++ b/scipy/_lib/_array_api.py
@@ -332,7 +332,7 @@ def xp_assert_equal(actual, desired, *, check_namespace=True, check_dtype=True,
     if xp is None:
         xp = array_namespace(actual)
     if isinstance(desired, list):
-        desired = xp.asarray(desired, dtype=actual.dtype)
+        desired = xp.asarray(desired, dtype=xp.asarray(actual).dtype)
 
     actual, desired = _strict_check(
         actual, desired, xp, check_namespace=check_namespace,
@@ -359,7 +359,7 @@ def xp_assert_close(actual, desired, *, rtol=None, atol=0, check_namespace=True,
     if xp is None:
         xp = array_namespace(actual)
     if isinstance(desired, list):
-        desired = xp.asarray(desired, dtype=actual.dtype)
+        desired = xp.asarray(desired, dtype=xp.asarray(actual).dtype)
```